src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp

```diff
@@ -724,6 +725,29 @@ QMCCostFunctionBatched::Return_rt QMCCostFunctionBatched::fillOverlapHamiltonian
 {
   ScopedTimer tmp_timer(fill_timer_);
 
+  hdf_archive hout;
+  if (output_matrix_inputs_hdf_)
```

Comment:
It seems like when this is set true if I have multiple ranks its going to be a problem.  Am I missing something.

Reply:
Yes, it's only for one rank.  I'm not sure how to handle multiple ranks, or if that's a case that's worth handling.

Reply:
Just realized if this only used with test programs it should removed from the input and the output argument set to true from the test program. Then some documentation in the code would suffice.

But if its is in the user facing input there must be a fatal error if used on multiple ranks or it needs to do something like put the rank number into the file name and be documented as writing for each rank.